docs/api/authorization.rst

```diff
@@ -0,0 +1,90 @@
+Authorization
+=============
```

Comment:
This is a new top-level page which comes after the introduction to the API, explaining when authorization is required and how it works before the reader gets to the API reference.